src/Microsoft.Identity.Extensions.Msal/MsalCacheStorage.cs

```
@@ -0,0 +1,474 @@
+                        }
+                    }
+
+                    _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, "After saving to linux keyring");
+                }
+
+                // Change data to 1 byte so we can write it to the cache file to update the last write time using the same write code used for windows.
+                data = new byte[] { 1 };
+            }
+
+            string directoryForCacheFile = Path.GetDirectoryName(CacheFilePath);
+            if (!Directory.Exists(directoryForCacheFile))
+            {
+                string directory = Path.GetDirectoryName(CacheFilePath);
+                _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, FormattableString.Invariant($"Creating directory '{directory}'"));
+                Directory.CreateDirectory(directory);
+            }
+
+            _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, FormattableString.Invariant($"Cache file directory exists. '{Directory.Exists(directoryForCacheFile)}' now writing cache file"));
+
+            TryProcessFile(() =>
+            {
+                File.WriteAllBytes(CacheFilePath, data);
+                _lastWriteTime = File.GetLastWriteTimeUtc(CacheFilePath);
+            });
+        }
+
+        private void ClearCore()
+        {
+            _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, "Clearing cache");
+            bool cacheFileExists = File.Exists(CacheFilePath);
+            _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, FormattableString.Invariant($"ReadDataCore Cache file exists '{cacheFileExists}'"));
+
+            TryProcessFile(() =>
+            {
+                    _logger.TraceEvent(TraceEventType.Information, /*id*/ 0, "Before deleting the cache file");
+                    try
+                    {
+                        File.Delete(CacheFilePath);
+                    }
+                    catch (Exception e)
```

Do you really want to catch all Exception types here?

---

There are 7 kinds of exceptions that can get thrown here, and they're not all nicely derived from a parent. Also we're simply logging and moving on, so unless there's a compelling reason to be more specific I'd prefer to leave it as is.